File: modules/sqlite-jdbc/src/main/java/io/roastedroot/sqlite4j/util/Logger.java
package io.roastedroot.sqlite4j.util;

import java.util.function.Supplier;
import org.jspecify.annotations.NonNull;

/** A simple internal Logger interface. */
public interface Logger {
    void trace(@NonNull Supplier<String> message);

    void info(@NonNull Supplier<String> message);

    void warn(@NonNull Supplier<String> message);

    void error(@NonNull Supplier<String> message, @NonNull Throwable t);
}

File: modules/sqlite-jdbc/src/main/java/io/roastedroot/sqlite4j/util/Logger.kt
package io.roastedroot.sqlite4j.util

import java.util.function.Supplier

/** A simple internal Logger interface.  */
interface Logger {
    fun trace(message: Supplier<String>)

    fun info(message: Supplier<String>)

    fun warn(message: Supplier<String>)

    fun error(message: Supplier<String>, t: Throwable)
}
